src/Config/SystemConfigReader.ts

```diff
@@ -0,0 +1,153 @@
+import { logger } from '../Utils';
+import request, { RequestPromiseAPI } from 'request-promise';
+import { promises, constants } from 'fs';
+import { join } from 'path';
+import { homedir } from 'os';
+
+const { readFile, writeFile, access } = promises;
+const { F_OK } = constants;
+
+export interface SystemConfig {
+  sentryDsn?: string;
+}
+
+interface SystemConfigFile {
+  data: SystemConfig;
+  updatedAt: Date;
+}
+
+export class SystemConfigReader {
+  private readonly rotationInterval = 3600000;
+  private readonly path = join(homedir(), '.brightclirc');
+  private client: RequestPromiseAPI;
+
+  constructor(baseUrl: string) {
+    this.client = request.defaults({
+      baseUrl,
+      timeout: 1500,
+      json: true
+    });
+  }
+
+  public async read(): Promise<SystemConfig> {
+    await this.rotateIfNecessary();
+    const configFile = await this.getConfigFile();
+
+    return {
+      sentryDsn: process.env['SENTRY_DSN'],
+      ...configFile.data
+    };
+  }
+
+  private async rotateIfNecessary() {
+    if (process.env.NODE_ENV !== 'production') {
+      return;
+    }
+
+    logger.debug('Trying to rotate system config');
+
+    const configFile = await this.getConfigFile();
+    const lifeTime = Date.now() - configFile.updatedAt.getTime();
+
+    if (lifeTime < this.rotationInterval) {
+      logger.debug(
+        'Rotation is not needed, system config is fresh: %s ms',
+        lifeTime
+      );
+
+      return;
+    }
+
+    logger.debug(
+      "Rotating system config because it's too old: %s ms",
+      lifeTime
+    );
+
+    const newConfig = await this.fetchNewConfig();
+
+    if (newConfig) {
+      await this.updateConfigFile({
+        data: newConfig,
+        updatedAt: new Date()
+      });
+    } else {
+      logger.debug('Rotation failed');
+
+      await this.updateConfigFile({
+        ...configFile,
+        updatedAt: new Date()
+      });
```

Comment:
Seems this is redundant. Should we update the configuration as though the rotation was successful?

Reply:
I think yes, just because if we don't do that, it will try to perform rotation on every `bright-cli` call causing at most 1.5 seconds delay.

Since rotation logic is not crucial for the cli for now, I think it's ok in case of failure to repeat it later.

Reply:
As you wish. However, I'm still uncertain about how it is supposed to work for long-running processes like the repeater which is intended to work for months.

Reply:
Let's have a separate PR for the background rotation

Reply:
@lsndr please open a ticket in Jira